app/services/IdentityService.scala

```diff
@@ -29,31 +27,36 @@ class IdentityService(identityApiClient: IdentityApiClient) extends LazyLogging
       (response.json \ "user").asOpt[IdUser]
     }
 
-  def registerGuest(personalData: PersonalData): Future[UserId] = {
-
+ def registerGuest(personalData: PersonalData): Future[GuestUser] = {
     val json = JsObject(Map(
       "primaryEmailAddress" -> JsString(personalData.email),
       "privateFields" -> JsObject(Map(
-        "firstName" -> JsString(personalData.firstName),
+        "firstName" ->  JsString(personalData.firstName),
         "secondName" -> JsString(personalData.lastName),
         "billingAddress1" -> JsString(personalData.address.address1),
         "billingAddress2" -> JsString(personalData.address.address2),
         "billingAddress3" -> JsString(personalData.address.town),
         "billingPostcode" -> JsString(personalData.address.postcode),
-        "billingCountry" -> JsString("United Kingdom")
-      ).toSeq),
-      "statusFields" -> JsObject(Map("receiveGnmMarketing" -> JsBoolean(true)).toSeq)
-    ).toSeq)
+        "billingCountry" -> JsString("United Kingdom"),
+        "statusFields" -> JsObject(Map("receiveGnmMarketing" -> JsBoolean(true)))
+    ))))
 
+    identityApiClient.createGuest(json).map(response => response.json.as[GuestUser])
+  }
 
-    identityApiClient.createGuest(json).map { response =>
-      UserId((response.json \ "guestRegistrationRequest" \ "userId").as[String])
+  def convertGuest(gu: GuestUser, password: String): Future[Unit] = {
+    val json = JsObject(Map("password" -> JsString(password)))
+    identityApiClient.convertGuest(json, gu).map { response =>
+      if (!response.status.toString.matches("""2\d\d""")) {
```

Comment:
I'm with @jennysivapalan on this one - I think just match against the Play constant `CREATED` (== `201`). If the return status code of this endpoint changes, I'd like to know why.